### PR TITLE
Initiated testing environment

### PR DESCRIPTION
**Pull request by filipowm**
Added:

- few basic tests
- `tox` to manage test environment
- `Makefile` to easily manage project tasks

### CODE REVIEW

**Comment by filipowm** [comment]
eh, we can't use GH actions on Roche org, so they fail:
`- Workflows can't be executed on this repository. Please check your payment method or billing status.` :/

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.PHONY: help clean-pyc clean-build isort lint test docker-build
```
Comment (randombenj) [inline Makefile:1]: For me this makefile is just one level of indirection to much. Why don't you just call tox directly in gitlab actions?

File: README.md [hunk 2]
```diff
@@ -12,16 +12,73 @@ Manage GitLab configuration as code to make it easily manageable and reproducibl
...
+
+
+### Using Docker image
 
 
 ### Examples
 
 
-## Development
+## Contributing
```
Comment (randombenj) [inline README.md]: Consider to move this to the `CONTRIBUTING` file.

File: README.md [hunk 3]
```diff
@@ -12,16 +12,73 @@ Manage GitLab configuration as code to make it easily manageable and reproducibl
...
+We use black as code formatter, so you'll need to format your changes using 
+the [black code formatter](https://github.com/python/black).
+
+Just run:
+```bash
+cd python-gitlab/
+pip3 install --user tox
+tox -e black
+```
+to format your code according to our guidelines (`tox` is required).
```
Comment (randombenj) [inline README.md]:
```suggestion
to format your code according to our guidelines ([tox](https://tox.readthedocs.io/en/latest/) is required).
```

File: test-requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+pytest
```
Comment (randombenj) [inline test-requirements.txt]: You may want to explicitly define the version of your test dependencies so when you update them, you have to do this manually and you know its not the dependencies that fail your tests but the tests themselves.

File: tests/data/yaml_include_txt.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# header1
```
Comment (randombenj) [inline tests/data/yaml_include_txt.md:1]: What is this?

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,41 @@
+[tox]
```
Comment (randombenj) [inline tox.ini:1]: Like it! :tada:

File: tests/data/yaml_include_txt.md (hunk 5 above)
Comment (filipowm) [reply tests/data/yaml_include_txt.md:1]: Just a test file to verify that `!include` directive will not only inject yamls into yaml, but also plain text files (as simple string)

File: test-requirements.txt (hunk 4 above)
Comment (filipowm) [reply test-requirements.txt]: Done

File: README.md (hunk 2 above)
Comment (filipowm) [reply README.md]: Done

File: Makefile (hunk 1 above)
Comment (filipowm) [reply Makefile:1]: Yes and no. For me tox is just a test manager, while makefile may contain way more helpful commands (also wrapping tox commands). This allows me to standardize commands, regardless of what I use behind, and also saves me (and others) from digging through "helper" shell scripts.

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at [INSERT EMAIL ADDRESS]. All
```
Comment (randombenj) [inline CODE_OF_CONDUCT.md]: insert mail address ;)

**Review by randombenj — APPROVED** [review]
Cool :tada:

File: CODE_OF_CONDUCT.md (hunk 7 above)
Comment (filipowm) [reply CODE_OF_CONDUCT.md]: Added brand new `global.open_source@roche.com` :-)

**Pull request merged**